### PR TITLE
Understanding Support Vector Machines

### LINKED ISSUE #777: Understanding support vector machines
**Brief Summary:** 

Support vector machines are set of supervised learning methods mainly used for classification problems though they can also be used for regression and outlier detection, they are very efficient in high dimensional spaces and are  memory efficient since they only use  a subset of the training points in the decision function

 **Key Takeaways:** 
1. Understand what support vector machines are.
2. Appreciating what support vector machines are  used for in machine learning.
3. Understand the matematical formulation behind support vector machines.
4. Be able to solve a real world problem using  support vector machines.

### _References:_ 
N/A

### PR DESCRIPTION
**Pull request by mykolkiprono**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

Fixes #777

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

File: articles/support-vector-machines/index.md
Comment (lalith1403) [inline articles/support-vector-machines/index.md:20]: Capitalization required: Mathematical formulation.

File: articles/support-vector-machines/index.md
Comment (lalith1403) [inline articles/support-vector-machines/index.md:64]: These equations aren't written correctly. Use \frac{}{} to represent fractions correctly,

File: articles/support-vector-machines/index.md
Comment (lalith1403) [inline articles/support-vector-machines/index.md:88]: The equations are looking bad because you aren't using Latex for all equations. Consider using Latex for the entire equation, including the summation and alpha signs.

Comment (lalith1403) [reply articles/support-vector-machines/index.md:88]: $$ L = \sum_{\alpha} \frac{a}{b}^\alpha $$

Comment (lalith1403) [reply articles/support-vector-machines/index.md:88]: Preview this in vscode. Online Markdown editors don;t support Latex equations.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Make those changes. Will later move onto the code. The theory is the main bottleneck here. SVMs are hard to explain. Take your time. Think of explaining the equations intuitively as well. Very important to deliver value to the reader.

**Comment by hectorkambow** [comment]
Thank you for submitting your 1st article to the EngEd program and welcome @mykolkiprono 
[Understanding Support Vector Machines #1007.pdf](https://github.com/section-io/engineering-education/files/5799227/Understanding.Support.Vector.Machines.1007.pdf)


Just wanted to note that your article raised some very minor flags on our 3rd party plagiarism checker (see attachement)
This is just a friendly reminder to please be sure we are adding proper sources within the article wherever applicable.
Let be sure to not use any sentences word for word - unless needed to make a point.

Even code samples should be unique and original (unless they are required frameworks or boilerplate code.)

Thank you 👍

**Comment by mykolkiprono** [comment]
@lalith1403 I have made the requested changes.

**Comment by lalith1403** [comment]
image
The entire latex code is wrong. I am not able to edit on my end. The reason being, the equations are convoluted. Can you check the entire article once? Also, preview the article before submitting it.

**Comment by mykolkiprono** [comment]
@lalith1403 the latex is correct preview it online.
Screenshot 2021-01-16 132441

**Comment by lalith1403** [comment]
@hectorkambow Can you check once if the equations are visible on your End? I am getting errors since I am not used to the authors way of typing equations.

**Comment by hectorkambow** [comment]
@lalith1403  

I tried to fix the mathjax scripts - they seem to be rendering on my end- just waiting for confirmation that the codes are accurate. 
And that the article is ready for final review.

**Comment by mykolkiprono** [comment]
@lalith1403 the code is working well.

**Comment by hectorkambow** [comment]
@mykolkiprono  👋  
I was previewing your article (how it would look on the site). 

It seems the equations are throwing the article off - they seem to be thrown into the article all over the place. 
(see image)
image

Is there anything that can be done to fix this - make a bit more neater before publishing. 

Some code snippets are not rendering either.

**Comment by lalith1403** [comment]
@mykolkiprono The article needs further refining. The equations are just mentioned. You need to explain the entire derivation process in detail. I also found a couple of Latex errors. Lines 46 and 47.

Online Markdown Editor - Dillinger, the Last Markdown Editor ever

I am assuming you are trying to derive the dual formulation of SVM. However, the reader needs to know why you are going for the dual formulation in the first place. Why is it necessary? To perform operations efficiently and introduce kernels. Next question is, what are kernels? Approach the problem logically and sequentially. Explain it like you would explain to an absolute beginner. Then you can talk about the derivation. 

Currently, it does not add up because the derivations are just equations. The equations convey meaning, which has to be conveyed to enhance readability. Also, refrain from using Then +ve, or such small phrases in your article. Explain concepts in sentences. 

The code works fine. We need to work on the derivation aspect of the article. Let me know once done. Also, attach a repl link so that the reader can run the code immediately. Include 3 repl links, and attach it as follows:

Run the code [here](https://repl.it/@lalithNarayan/ThoseSlimWatchdog#main.py)

One more round of revision and this article will be good to go.

**Comment by mykolkiprono** [comment]
@lalith1403 I've made the requested changes.

**Comment by hectorkambow** [comment]
@mykolkiprono  

Upon a quick glance your article still has some issue with misspellings and formatting. 

I'll be closing this for now - so you can try to fix the issues on your forked repo - 👍

**Comment by mykolkiprono** [comment]
@hectorkambow I have made the changes.

**Pull request closed without merging**
